src/base/bittorrent/torrentimpl.cpp

```diff
@@ -254,6 +240,19 @@ namespace
         status.pieces = params.have_pieces;
         status.verified_pieces = params.verified_pieces;
     }
+
+    namespace LT
+    {
+        constexpr lt::download_priority_t toNative(const DownloadPriority priority) noexcept
+        {
+            return static_cast<lt::download_priority_t>(static_cast<std::underlying_type_t<DownloadPriority>>(priority));
+        }
+
+        constexpr DownloadPriority fromNative(const lt::download_priority_t priority) noexcept
+        {
+            return static_cast<DownloadPriority>(toLTUnderlyingType(priority));
+        }
+    }
```

Comment:
@Chocobo1 
Do you have any ideas about the best place for such functions?
I can put them in `downloadpriority.h`, but I don't like that it will have to pollute "the outside world" with `libtorrent` stuff (although based on our recent discussion, it seems that you don't feel this is a problem).
Personally, it is not a problem for me to have them in the implementation files, even taking into account the fact that they are duplicated. These are just trivial helpers to shorten the code a bit.

Reply:
I would try to de-duplicate them and then worry about the pollution problem.
maybe move `ltunderlyingtype.h` to `lttypes.h` and put it in there.